### ISSUE TITLE
Project: A smarter meeting availability scheduling API

### ISSUE THREAD

**Issue by jamesmcroft** [microsoft/hack-together#63]
### Project name

Meeting Scheduler

### Description

The project is an Azure Functions API that sits on top of the Microsoft Graph APIs that can check calendars of multiple users within a Microsoft 365 tenant for a specific duration within a period (start and end time) and find availability for the best time for a potential meeting. 

This request takes into consideration each participant's schedules, as well as including everyone's time zone to ensure that the times available are most convenient for everyone.

### Repo URL

https://github.com/jamesmcroft/msgraph-meeting-scheduler

### Team members

jamesmcroft

**Comment by waldekmastykarz** [comment]
Awesome! Thank you for sharing your hack with us!

**Comment by jamesmcroft** [comment]
Not a problem at all @waldekmastykarz! I have been meaning to explore the Microsoft Graph APIs to work out how I can take advantage of them for client projects at work and this was a great opportunity for me to focus.

**Comment by svrooij** [comment]
I see some nice badges in the readme, looking good!

Since your code seems to be utilizing data from just one tenant, you could make it even better by skipping the app registration all together and just setup the Azure Functions to allow Graph access through it's managed identity.

Check out this post https://svrooij.io/2022/04/21/access-api-with-managed-identity/

**Comment by jamesmcroft** [comment]
Good shout! I've not got this up in Azure just yet but this sounds like a solid plan! 

Will get this sorted and pop out a URL for it!

**Comment by waldekmastykarz** [comment]
Hey @jamesmcroft, would you mind recording a short video walkthrough of your hack? Any format will work. It's not required for the hackathon but it would help us showcase your work both internally at Microsoft and publicly in the community. 😊

**Comment by jamesmcroft** [comment]
@waldekmastykarz of course, not a problem at all! I'm still planning on continuing this project post hack building a little UI, potentially something low code that can integrate into Team around it. 

I'll get something sorted and shared with you here 😄

**Comment by waldekmastykarz** [comment]
Brilliant! Thank you so much and looking forward to the future upgrades of your hack!

**Comment by jamesmcroft** [comment]
[Video and commentary around Hackathon submission](https://1drv.ms/v/s!AtBQjsPaJ0IRyJEABGCJxY8l5xrmfw?e=ZteE39).
